openmdao/devtools/xdsm_viewer/xdsm_writer.py

```diff
@@ -1075,22 +1105,18 @@ def _prune_connections(conns, model_path=None, sep='.'):
         return conns, external_inputs, external_outputs
     else:
         for conn in conns:
-            src = src0 = conn['src']
-            if src.startswith(model_path):
-                src = src[len(model_path):]
-            src_path = _format_name(src.rsplit(sep, 1)[0])
-            tgt = tgt0 = conn['tgt']
-            if tgt.startswith(model_path):
-                tgt = tgt[len(model_path):]
-            tgt_path = _format_name(tgt.rsplit(sep, 1)[0])
-
-            if src0.startswith(model_path) and tgt0.startswith(model_path):
+            src = conn['src']
+            src_path = _make_rel_path(src, model_path=model_path)
+            tgt = conn['tgt']
+            tgt_path = _make_rel_path(tgt, model_path=model_path)
+
+            if src.startswith(model_path) and tgt.startswith(model_path):
```

Comment:
same problem here as mentioned before.  You need to check startswith using model_path + '.' or you can get false matches.

Reply:
indeed, nice catch @naylor-b